src/content/insertable-streams/webgpu/js/multi_video_main.js

```diff
@@ -0,0 +1,274 @@
+                ' Please see the note at the bottom of the page.';
+            document.getElementById('errorMsg').innerText = errorMessage;
+            console.log(errorMessage);
+            return;
+        }
+        this.context_ = context;
+        const adapter = await navigator.gpu.requestAdapter();
+        const device = await adapter.requestDevice();
+        this.device_ = device;
+        if (this.device_ === null || this.device_ === undefined) {
+            console.log('[WebGPUTransform] requestDevice failed.')
+            return;
+        }
+        const swapChainFormat = 'bgra8unorm';
+
+        const rectVerts = new Float32Array([
+            1.0, 1.0, 0.0, 1.0, 0.0,
+            1.0, -1.0, 0.0, 1.0, 1.0,
+            -1.0, -1.0, 0.0, 0.0, 1.0,
+            1.0, 1.0, 0.0, 1.0, 0.0,
+            -1.0, -1.0, 0.0, 0.0, 1.0,
+            -1.0, 1.0, 0.0, 0.0, 0.0,
+        ]);
+        // Creates a GPU buffer.
+        const vertexBuffer = device.createBuffer({
+            size: rectVerts.byteLength,
+            usage: GPUBufferUsage.VERTEX,
+            mappedAtCreation: true,
+        });
+        // Copies rectVerts to vertexBuffer
+        new Float32Array(vertexBuffer.getMappedRange()).set(rectVerts);
+        vertexBuffer.unmap();
+        this.vertexBuffer_ = vertexBuffer;
+
+        context.configure({
+            device,
+            format: swapChainFormat
+        })
+
+        this.renderPipeline_ = device.createRenderPipeline({
+            vertex: {
+                module: device.createShaderModule({
+                    code: wgslShaders.vertex,
+                }),
+                entryPoint: 'main',
+                buffers: [
+                    {
+                        arrayStride: 20,
+                        attributes: [
+                            {
+                                // position
+                                shaderLocation: 0,
+                                offset: 0,
+                                format: 'float32x3',
+                            },
+                            {
+                                // uv
+                                shaderLocation: 1,
+                                offset: 12,
+                                format: 'float32x2',
+                            },
+                        ],
+                    },
+                ],
+            },
+            fragment: {
+                module: device.createShaderModule({
+                    code: wgslShaders.fragment,
+                }),
+                entryPoint: 'main',
+                targets: [
+                    {
+                        format: swapChainFormat,
+                    },
+                ],
+            },
+            primitive: {
+                topology: 'triangle-list',
+            },
+        });
+
+        this.videoTexture_ = device.createTexture({
+            size: {
```

Comment:
nit: texture defaults mean you only need to specify usage / format / size. And you can do `size: [480 * 2, 270 * 2]` too!